astropy/utils/data_info.py

```diff
@@ -355,6 +355,9 @@ def __set__(self, instance, value):
                 # _attrs, so speed matters up by not accessing defaults.
                 # Doing this before difference in for loop helps speed.
                 attr_names = attr_names & set(value._attrs)  # NOT in-place!
+            else:
+                # For different classes, copy over the attributes in common.
+                attr_names = attr_names & (value.attr_names - value._attrs_no_copy)
```

Comment:
I'm not sure how we have avoided to run into this before, but without this one cannot transfer info from a `Quantity` to a `MaskedQuantity`, since the former does not actually have `serialize_methods`.

Reply:
To clarify the relation to #11917: without this change one cannot do `masked_column.info = column.info`.